cardano-db-sync/src/Cardano/DbSync/Plugin/Epoch.hs

```diff
@@ -191,8 +243,10 @@ updateChainTipEpochVar _trce = do
     currentTime <- getCurrentTime
     case eMeta of
       Left _ -> do
+        -- This should be propagated upwards, not swallowed!
         atomicWriteIORef latestChainTipEpochVar 0
       Right meta -> do
+        -- Weird.
```

Comment:
It may need a comment or better yet be moved to a function with a good name *and* a comment but what makes it weird? It estimates the current top epoch based on the current time, the chain start time, the slot interval and the number of slots per epoch.

Reply:
Right, but I'm not sure if we actually have the correct time synced up in all the node instances? Why does the current top epoch of my local node match the one I'm connected to?
Don't we always have some (small) time discrepancy and should we even rely on time as a measure for anything in the blockchain world?

Reply:
Validation of the chain *requires* the system clock to be mostly correct (ie within a second or two), because the prorocol requires that someone who forges a block at time X is actually the valid slot leader for that slot.

We can therefore assume that the clock is more than close enough to estimate the current epoch.